c7n/resources/rds.py

```diff
@@ -1568,3 +1568,105 @@ def process(self, resources, event=None):
                     results.append(resource)
                     break
         return results
+
+
+@actions.register('modify-db')
+class ModifyDb(BaseAction):
+    """Modifies an RDS instance based on specified parameter
+    using ModifyDbInstance.
+
+    'Update' is an array with with key value pairs that should be set to
+    the keyword and value you wish to modify.
+    'Immediate" determines whether the modification is applied immediately
+    or not. If 'immediate' is not specified, default is false.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: disable-rds-deletion-protection
+                resource: rds
+                filters:
+                  - DeletionProtection: true
+                  - PubliclyAccessible: true
+                actions:
+                  - type: modify-db
+                    update:
+                        - keyword: 'DeletionProtection'
+                            value: false
+                        - keyword: 'PubliclyAccessible'
+                            value: false
+                    immediate: true
+    """
+
+    schema = type_schema(
+        'modify-db',
+        immediate={"type": 'boolean'},
+        update={
+            'type': 'array',
+            'items': {
+                'type': 'object',
+                'properties': {
+                    'keyword': {'type': 'string', 'enum': [
```

Comment:
s/keyword/property

Reply:
Unclear what you mean

Reply:
i'm suggesting renaming 'keyword' to 'property' or even 'attribute'. keyword is not the right name for what's being set on the instance.

Reply:
changed